qlik-oss/sn-scatter-plot
# fix: animations add options and improvements

## Descriptions
1. Enable animations at chart level. 
    Add `options.chartAnimations` check. That option depends on sense-client animations toggle switch and `CHART_ANIMATIONS_TOGGLE`
2. Remove updating stages when main component has no updating node
3. Disable animations when resizing
4. Disable animations when changing properties (constraints active is true).
5. Disable animations when partial updating (zooming).
6. Allow animations in agnostic chart.

**Some specific things for scatter plot:**

7. Disable animations when rendering progressively
8. Disable animations when changing from heatmap to points and vice verse, since the new main component (point/heatmap) will appear immediately while other components are being animated and it does not look good.

## Verification
1. When the CHART_ANIMATIONS_TOGGLE is on, animations should be on by default. After that, the toggle button should work and its value if saved should remain after refreshing. If the CHART_ANIMATIONS_TOGGLE flag is removed, animations should not work, even if the saved stage of the toggle is on.

    https://user-images.githubusercontent.com/70384379/201362233-764e629e-03e0-4d33-904e-5a49734b004b.mov

2. Remove updating stages when main component has no updating node

    https://user-images.githubusercontent.com/70384379/201362828-61d9d688-94b0-4654-b4fe-6a0a0226a62d.mov

3. Disable animations when resizing

    https://user-images.githubusercontent.com/70384379/201363136-5f50e712-a964-476c-9823-59aba9418a9f.mov

4. Disable animations when changing properties (constraints active is true).

    https://user-images.githubusercontent.com/70384379/201363331-570b26fb-c258-442e-87da-dcbbcf1a76fa.mov

5. Disable animations when zooming.

    https://user-images.githubusercontent.com/70384379/201363899-3c7457aa-1ac7-4bee-832c-34ea61f26632.mov

6. Allow animations in agnostic chart.

    https://user-images.githubusercontent.com/70384379/201674738-c1229acb-4f3f-4388-89d4-1dcb3b2219af.mov

8. Disable animations when changing from heatmap to points and vice verse

    https://user-images.githubusercontent.com/70384379/201364174-cc0a4bfa-b2a8-4dd0-b2e9-66a3a2409588.mov
